grouper/stats.py

```diff
@@ -1,24 +1,43 @@
-from tornado.process import task_id
-
 from grouper.plugin import get_plugin_proxy
 
 
-def set_defaults():
-    # type: () -> None
-    instance = task_id()
-    if instance is None:
-        instance = 0
+def log_background_run(success):
```

Comment:
Delete all of these (and I think you may be able to get rid of the entire file as a result). All new code should call the plugin proxy directly that it was configured with as part of initialization. Old code should call `get_plugin_proxy()` directly. There shouldn't be any need for this additional level of indirection.